connector/src/main/scala/com/basho/riak/spark/util/TSConversionUtil.scala

```diff
@@ -91,10 +91,10 @@ object TimeSeriesToSparkSqlConversion {
 
   def asSparkSchema(columns: Seq[ColumnDescription]): StructType = StructType(columns.map(asStructField))
 
-  private def validateSchema(schema: StructType, columns: Seq[String]): Unit = {
-    val msg = "Provided schema does not match the riak row columns"
-    require(schema.length == columns.length, msg)
-    require(schema.forall(sf => columns.contains(sf.name)), msg)
+  private def validateSchema(schema: StructType, columns: Seq[String]): Unit = schema.map(_.name).diff(columns) match {
```

Comment:
Actually there is no need to check types here. In case of unmatched types exception will be thrown in one of following methods: `com.basho.riak.spark.util.TimeSeriesToSparkSqlConversion#cellValue` and `com.basho.riak.spark.util.TimeSeriesToSparkSqlConversion#asDataType`